src/libcore/sync/atomic.rs

```diff
@@ -1550,12 +1550,30 @@ unsafe fn atomic_xor<T>(dst: *mut T, val: T, order: Ordering) -> T {
 
 /// An atomic fence.
 ///
+/// A fence creates happens-before edges between other atomic operations or
+/// fences when the certain condition is met. Also, a fence prevents reordering
+/// of operations around a fence by CPU or a compiler. Each behavior is
+/// controlled by the value of [`Ordering`].
+///
 /// A fence 'A' which has [`Release`] ordering semantics, synchronizes with a
```

Comment:
Perhaps this should say "A fence 'A' which has (at least) [`Release`]..."? Note the text in parantheses.